app/controllers/github_hook_controller.rb

```diff
@@ -1,7 +1,7 @@
 require "json"
 
 class GithubHookController < ApplicationController
-  skip_before_filter :verify_authenticity_token, :check_if_login_required
+  skip_before_action :verify_authenticity_token, :check_if_login_required
```

Comment:
To maintain compatibility, maybe change
```
skip_before_action :verify_authenticity_token, :check_if_login_required
```
to:
```
skip_before = respond_to?(:skip_before_action) ? :skip_before_action : :skip_before_filter
send skip_before, :verify_authenticity_token, :check_if_login_required
```

Reply:
What's the status of Redmine versions these days? Looks like 3.4 and 4.0 are the official releases, and going through the changelogs seems to indicate the following Rails versions are used:

| Redmine 4.0 | Rails 5 |
| Redmine 3.4 | Rails 4.2.11 |
| Redmine 3.3 | Rails 4.2.11 |
| Redmine 3.2 | Rails 4.2.3 |
| Redmine 3.1 | Rails 4.2.4 |
| Redmine 3.0 | Rails 4.2.3 |

So if I am reading https://github.com/rails/rails/pull/18410 right, every Redmine version in the 3.x line should work with both `skip_before_action` and `skip_before_filter` with the latter giving a warning. Redmine 4.x only works with `skip_before_action`.

This also means that there is no officially supported release of Redmine that requires `skip_before_filter`, seeing how that would be Redmine 2.x and earlier, correct?

In that case, I'd say it makes sense to dump the backwards compatibility here and bump the gem release to a major version, as you suggest @slamotte. While the change in #93 is clever and functional, it's also a bit of somewhat obscure metaprogramming that I'd prefer to not have to drag along for the future.

Reply:
Agreed. Anyone on Redmine 2.x can just use the older gem version.

BTW thanks for creating this gem, it's great!